### ISSUE TITLE
ParamSet dependencies are not properly deep cloned

### ISSUE THREAD

**Issue by pfistfl** [mlr-org/paradox#271]
I guess the following should work:

```r
  library(paradox)
  ps = ParamSet$new()
  ps$add(ParamFct$new("a", levels = letters[1:3]))
  ps$add(ParamFct$new("b", levels = letters[1:3]))
  ps$add_dep("a", "b", CondAnyOf$new(letters[1:2]))

  ps2 = ps$clone(deep = TRUE)
  expect_deep_clone(ps, ps2)
```

I get the following error:
Error: Object addresses differ at path ROOT->[element .__enclos_env__] '.__enclos_env__'->[element private] 'private'->[element .deps] '.deps'->[element cond] 'cond'->[element 1] isn't true.

`expect_deep_clone` is from [here](https://github.com/mlr-org/mlr3pipelines/blob/master/tests/testthat/helper_functions.R)

**Comment by mb706** [comment]
Also addressed by #273 now

**Comment by jakob-r** [comment]
close because #273 is merged
